src/apprt/embedded.zig

```diff
@@ -1346,13 +1347,17 @@ pub const CAPI = struct {
         opts: *const apprt.runtime.App.Options,
         config: *const Config,
     ) !*App {
-        var core_app = try CoreApp.create(global.alloc);
-        errdefer core_app.destroy();
+        var core_app = try global.alloc.create(CoreApp);
+        try core_app.init(global.alloc);
+        errdefer {
+            core_app.deinit();
+            global.alloc.destroy(core_app);
```

Comment:
This part of the error defer has to be above the init so if the init fails then it will be destroyed. The ordering will still be correct as the second errdefer runs first see this as an example 
```zig
const std = @import("std");

fn fail() !void {
    return error.Fail;
}
pub fn main() !void {
    errdefer std.debug.print("first\n", .{});
    errdefer std.debug.print("second\n", .{});
    try fail();
}
```